### ISSUE TITLE
Unable to calculate height of elements with images in Safari

### ISSUE THREAD

**Issue by Jacob87** [liabru/jquery-match-height#58]
I'm unsure whether this has been discussed before, as I see quite a few issues related to Safari, however I want to at least expand on the previous issues by reporting some findings.

This plugin is made to initialise its API using the `jQuery(document).ready()` function and then run the internal `_update()` function when `jQuery(window).load()` is fired. According to [this article](http://tech-blog.maddyzone.com/javascript/document-ready-vs-window-load-vs-window-onload) it seems that images are only fully loaded after `jQuery(window).load()` is fired. So this plugin is initialised too soon when calculating the height for elements containing images, but according to previous issues, this edge case should be fixed by running an `_update()` when the 'load' event is fired, which is apparent in the source code, but here is the issue: Safari does not update properly after the first calculation has been applied as [illustrated in this Screencast video](http://screencast.com/t/pEB0x63KDf0v). I have to invoke `matchHeight('remove')` first and then reapply matchHeight to get Safari to work properly. According to further tests, it seems, that moving the initial call to `$(matchHeight._applyDataApi);` inside the 'load' event fixes the issue, although that approach may cause unnecessary delays for other use cases. The only fix for me right now is to remove matchHeight and then reapply in the 'load' event. However, this will cause serious "bumpyness" for the visitors if the height of the images varies too much. What to do?

**Comment by liabru** [comment]
Would you be able to provide a jsfiddle that demonstrates the issue and another with your fix applied? Thanks

**Comment by Jacob87** [comment]
Yes, sorry for the delay in answering your comment. I'll try and set up a jsfiddle for you, however that might not be before the week after next. I hope the issue will remain open until then. Thanks!

**Comment by Jacob87** [comment]
Btw., I must admit that I cannot get it to fail using the latest master, but it requires more testing to be sure that I did not miss anything.

**Comment by Jacob87** [comment]
My testing has concluded. I found that the issue with Safari arose when the `data-mh` handles was placed on sub-elements rather than sibling elements. Moving the `data-mh` handles to sibling elements (parents of the former mentioned sub-elements), and an upgrade to the latest master, made Safari work again. Unfortunately I am uncertain whether it was the upgrade or the moving of the attributes, that did the trick in the end. However, putting the `data-mh` attributes on the specific elements that should be aligned with each other - not on their sub-elements - seems to be the correct way of doing things semantically anyway.

To summarize, this does **not** work:
``` html
<div class="item">
    <div data-mh="item">...</div>
</div>
```

This **works**:
``` html
<div class="item" data-mh="item">
    <div>...</div>
</div>
```

**Comment by liabru** [comment]
Thanks for providing an update on your progress and I'm glad you managed to resolve it. You're right that the data attributes should be used on the exact elements you wish to match (not their parents or children).
